### PR TITLE
Add Trigger Campaign Functionality 

### PR DESCRIPTION
**Pull request by marcelolebre**
This PR adds the API trigger for campaigns as stated here: [https://customer.io/docs/api/#apibroadcast_trigger](https://customer.io/docs/api/#apibroadcast_trigger)

### CODE REVIEW

File: lib/customerio.ex [hunk 1]
```diff
@@ -801,4 +801,24 @@ defmodule Customerio do
...
+    * `data` - data to be included in the campaign.
+
+  ## Example
+
+  ```elixir
+  iex> Customerio.trigger_campaign(1, %{data: %{title: "hello"}})
+  ```
+  """
+
+  def trigger_campaign(id, data \\ []) do
```
Comment (Virviil) [inline lib/customerio.ex]: API has additional data, but data seems to be a map, not a list(Keyword).
May be it'll be great idea to add a test with additional params, to ensure that they are passed in a right way.

Also, each function in library has opportunity to controll hackney with additional params.
Let's keep this opportunity.

```suggestion
  def trigger_campaign(id, data_map, opts \\ []) do
```

File: lib/customerio.ex [hunk 2]
```diff
@@ -801,4 +801,24 @@ defmodule Customerio do
...
+
+    * `data` - data to be included in the campaign.
+
+  ## Example
+
+  ```elixir
+  iex> Customerio.trigger_campaign(1, %{data: %{title: "hello"}})
+  ```
+  """
+
```
Comment (Virviil) [inline lib/customerio.ex]: Don't forget to add specs for this function - dializer will make a great job here.

```suggestion

@spec ...
```

File: lib/customerio/util.ex [hunk 3]
```diff
@@ -20,6 +22,14 @@ defmodule Customerio.Util do
   """
   @type method :: :get | :post | :delete | :put | :patch
 
+  def send_behavioral_request(method, route, data_map, opts \\ []) do
```
Comment (Virviil) [inline lib/customerio/util.ex:35]: Let's add some specs to these methods.

```suggestion
@spec ...
  def send_behavioral_request(method, route, data_map, opts \\ []) do
```

File: lib/customerio/util.ex [hunk 4]
```diff
@@ -20,6 +22,14 @@ defmodule Customerio.Util do
...
+    send_request(method, @behavioral_base_route <> route, data_map, opts)
+  end
+
+  def send_api_request(method, route, data_map, opts \\ []) do
+    send_request(method, @api_base_route <> route, data_map, opts)
+  end
+
   @doc """
```
Comment (Virviil) [inline lib/customerio/util.ex:39]: `send_request` function seems to be called now only from inside this module. Probably, it should be marked as private?

**Review by Virviil — CHANGES_REQUESTED** [review]

File: lib/customerio/util.ex (hunk 4 above)
Comment (marcelolebre) [reply lib/customerio/util.ex:39]: Agreed

File: lib/customerio/util.ex (hunk 3 above)
Comment (marcelolebre) [reply lib/customerio/util.ex:35]: 👍

File: lib/customerio.ex (hunk 2 above)
Comment (marcelolebre) [reply lib/customerio.ex]: 👍

**Comment by marcelolebre** [comment]
All done, lmk what you think

**Comment by Virviil** [comment]
Great job!

**Pull request merged**